egs/wsj/s5/steps/dict/select_prons_bayesian.py

```diff
@@ -0,0 +1,447 @@
+    print('---------------------------------------------------------------------------------------------------')
+    print ('Num. words in the learned lexicon: {0} num. selected prons: {1}'.format(len(learned_lexicon), num_prons_tot))
+    print ('{0} selected prons came from reference candidate prons; {1} came from G2P candidate prons;'
+           '{2} came from phone-decoding candidate prons.'.format(reference_selected, g2p_selected, phone_decoding_selected)) 
+    return learned_lexicon
+
+def WriteEditsAndSummary(args, learned_lexicon, ref_lexicon, phone_decoding_lexicon, g2p_lexicon, counts, stats):
+    # Note that learned_lexicon and ref_lexicon are dicts of sets of prons, while the other two lexicons are sets of (word, pron) pairs.
+    thr = 3
+    words = [defaultdict(set) for i in range(4)] # "words" contains four bins, where we
+    # classify each word into, according to whether it's count > thr,
+    # and whether it's OOVs w.r.t the reference lexicon.
+
+    src = {}
+    print("# Note: This file contains pronunciation info for words who have candidate"
+          "prons from G2P/phone-decoding accepted in the learned lexicon."
+          ", sorted by their counts in acoustic training data, "
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 1st Col: source of the candidate pron: G(2P) / P(hone-decoding) / R(eference)."
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 2nd Col: soft counts from lattice-alignment (not augmented by prior-counts)."
+          ,file=args.ref_lexicon_edits_handle)
+    print("# 3rd Col: accepted or not in the learned lexicon (Y/N).", file=args.ref_lexicon_edits_handle)
+    print("# 4th Col: the pronunciation cadidate.", file=args.ref_lexicon_edits_handle)
+    
+    # words which are to be printed into the edits file.
+    words_to_edit = [] 
+    for word in learned_lexicon:
+        count = counts.get(word, 0)
+        flags = ['0' for i in range(3)] # "flags" contains three binary indicators, 
+        # indicating where this word's pronunciations come from.
+        for pron in learned_lexicon[word]:
+            if word in phone_decoding_lexicon and pron in phone_decoding_lexicon[word]:
+                flags[0] = '1'
+                src[(word, pron)] = 'P'
+            if word in ref_lexicon and pron in ref_lexicon[word]:
+                flags[1] = '1'
+                src[(word, pron)] = 'R'
+            if word in g2p_lexicon and pron in g2p_lexicon[word]:
+                flags[2] = '1'
+                src[(word, pron)] = 'G'
+        if word in ref_lexicon:
+            all_ref_prons_accepted = True
+            for pron in ref_lexicon[word]:
+                if pron not in learned_lexicon[word]:
+                    all_ref_prons_accepted = False
+                    break
+            if not all_ref_prons_accepted or flags[0] == '1' or flags[2] == '1':
+                words_to_edit.append((word, counts[word]))
+            if count > thr:
```

Comment:
please use threshold, not thr-- google style guide is against abbreviated names.